src/kernel/src/time.rs

```diff
@@ -1,23 +1,27 @@
-use alloc::{
-    sync::Arc,
-    vec::Vec
-};
+use alloc::{sync::Arc, vec::Vec};
 
 use twizzler_abi::syscall::{ClockInfo, FemtoSeconds};
 
-use crate::spinlock::Spinlock;
+use crate::{clock::Nanoseconds, spinlock::Spinlock};
 
 pub struct Ticks {
     pub value: u64,
-    pub rate: FemtoSeconds
+    pub rate: FemtoSeconds,
+}
+
+impl From<Ticks> for Nanoseconds {
+    fn from(t: Ticks) -> Self {
+        t.value * t.rate.0
+    }
 }
```

Comment:
This conversion is off by an order of magnitude, since `rate` is in `FemtoSeconds`. You should be able to convert the value by either manually dividing the right offset or doing something like this:
```rust
let nanos: NanoSeconds = (t.value * t.rate).into();
nanos.0
```
Here is an example in the `twizzler-abi` crate:
https://github.com/twizzler-operating-system/twizzler/blob/174a0ec9a4688e2e6bd5497670f2b195c5d16e4d/src/lib/twizzler-abi/src/syscall/time/units.rs#L158-L161

Reply:
Wait it might be simpler to do this:
```rust
// t.value * t.rate returns `TimeSpan`
// `TimeSpan` can be converted to nanoseconds (u128)
// `Nanoseconds` is u64
(t.value * t.rate).as_nanos() as Nanoseconds
```

Reply:
Ah yes, good catch. Some of this code is still working off the kernel's old Nanoseconds type. In a different patch, I'll replace it with the one from twizzler-abi.